Source/Dafny/Compilers/Compiler-java.cs

```diff
@@ -501,11 +501,20 @@ protected class ClassWriter : IClassWriter {
       }
       return wGet;
     }
+
+    private void AddTestCheckerIfNeeded(Declaration decl, ConcreteSyntaxTree wr) {
+      if (DafnyOptions.O.RunAllTests || !Attributes.Contains(decl.Attributes, "test")) {
+        return;
+      }
+      wr.WriteLine("@org.junit.jupiter.api.Test");
```

Comment:
`org.junit.jupiter.api.Test` is very opinionated. Can you please justify this choice?

Reply:
Certainly.  JUnit is a popular choice for programmers using build tools such as Gradle and Maven.  In fact, Gradle's own [documentation](https://docs.gradle.org/current/userguide/java_testing.html#using_junit5) recommends the use of JUnit 5.

Since JUnit 5 has extensive support for test assertions, it provides a natural "landing place" for the compilation of `expect` statements.  I can conceive a future instance where an `expect` statement gets turned into `assertDoesNotThrow` a `DafnyHaltException`.

Reply:
If anything, JUnit is more uncontroversially the most popular testing framework for Java, whereas XUnit has a much thinner margin over NUnit or MSTest. :)

Reply:
Is this a fair enough justification?